### PR TITLE
Warn users when using default pin factory for Servos and Distance Sensors, close #780

### LINKED ISSUE #780: Warn users when using default pin factory for servos
When someone uses the Servo class with gpiozero using the default settings (i.e. default pin factory RPi.GPIO) they get a bad experience and don't know that it's easily fixable, by using the pigpio pin factory.

We should use a warning (like we do with e.g. spi fallback) when someone tries to use Servo without pigpio.

@evildmp

### PR DESCRIPTION
**Pull request by SofiiaKosovan**
#780

### CODE REVIEW

**Comment by bennuttall** [comment]
Thank you @SofiiaKosovan!

Assigning to @waveform80 to review this

File: gpiozero/exc.py [hunk 1]
```diff
@@ -192,7 +192,10 @@ class SPIWarning(GPIOZeroWarning):
     "Base class for warnings related to the SPI implementation"
 
 class SPISoftwareFallback(SPIWarning):
-    "Warning raised when falling back to the software implementation"
+    "Warning raised when falling back to the SPI software implementation"
+
+class PWMSoftwareFallback(SPIWarning):
```
Comment (lurch) [inline gpiozero/exc.py]: I'm not sure if it (logically) makes sense for `PWMSoftwareFallback` to be a subclass of `SPIWarning` ? :man_shrugging:

File: gpiozero/output_devices.py [hunk 2]
```diff
@@ -1528,6 +1538,13 @@ def __init__(
             ),
             pin_factory=pin_factory
         )
+        
+        if PiGPIOFactory is None or not isinstance(self.pin_factory, PiGPIOFactory):
+            warnings.warn(PWMSoftwareFallback(
+                'To reduce servo jitter, use the pigpio pin factory.'
+                'See https://gpiozero.readthedocs.io/en/stable/api_output.html#servo for more info'
```
Comment (lurch) [inline gpiozero/output_devices.py]: It seems that the documentation being pointed to doesn't mention PiGpio at all? Maybe this PR should also update the associated documentation?

File: gpiozero/exc.py (hunk 1 above)
Comment (bennuttall) [reply gpiozero/exc.py]: True

File: gpiozero/output_devices.py (hunk 2 above)
Comment (bennuttall) [reply gpiozero/output_devices.py]: A note was added to the Servo docstring in this PR :)

**Pull request merged**